### DISCUSSION TITLE
How to configure the session time in QfieldCloud?

### DISCUSSION THREAD

**Discussion by rosaguilar** [opengisch/QField#5088] [Q&A]
I am working mostly offline with very slow internet. How can I configure Qfield to remain logged in. When the session has expired the value of @cloud_username is empty.

**Comment by suricactus** [comment]
Hey @rosaguilar . Thanks for reporting this issue, find it as an issue here: https://github.com/opengisch/QField/issues/5116

The problem is that `@cloud_username` and `@cloud_email` are only available when the user has successfully logged in. As a special present to you, I have prepared a PR that should fix your issue, please try the APKs from https://github.com/opengisch/QField/pull/5117 . For any further discussion, please move it to #5116 .

---

As for the original question how to remain QField users to remain logged in, there is no configuration for QField, but the cloud server. You can increase the session token expiration. It is by default configured to be 30 days. If you use local QFieldCloud installation, check the setting `QFIELDCLOUD_AUTH_TOKEN_EXPIRATION_HOURS` in your `settings.py`. On app.qfield.cloud the session will expire in 720 hours or 30 days:
https://github.com/opengisch/qfieldcloud/blob/master/.env.example#L107-L109

**Reply by rosaguilar** [reply]
Thank you so much for explaining where to set up and fixing the issue. You guys rock!
